dali/kernels/imgproc/paste/paste.h

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) 2021, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_KERNELS_IMGPROC_PASTE_PASTE_H_
+#define DALI_KERNELS_IMGPROC_PASTE_PASTE_H_
+
+#include <utility>
+#include "dali/util/ocv.h"
+#include "dali/core/common.h"
+#include "dali/core/convert.h"
+#include "dali/core/geom/box.h"
+#include "dali/core/error_handling.h"
+#include "dali/kernels/kernel.h"
+#include "dali/kernels/imgproc/roi.h"
+
+#define X_AXIS 0
+#define Y_AXIS 1
+#define C_AXIS 2
+
+namespace dali {
+namespace kernels {
+
+template<typename OutputType, typename InputType>
+class PasteCpu {
+ public:
+  using Image = InTensorCPU<InputType, 3>;
+  using OutImage = OutTensorCPU<OutputType>;
+  using Coords = InTensorCPU<const int, 1>;
+
+  KernelRequirements
+  Setup(KernelContext &context, const OutTensorCPU<OutputType> &pasteFrom, const Coords &in_anchors,
+        const Coords &in_shapes, const Coords &out_anchors) {
+    // Kernel is not aware of the size of the output. This should be handled in the operator.
+    return nullptr;
+  }
+
+
+  /**
+   * Assumes HWC memory layout
+   *
+   * @param out         Assumes, that memory is already allocated
+   * @param pasteFrom   Input image data.
+   * @param in_anchors  1D int view denoting where the selected part starts.
+   * @param in_shapes   1D int view denoting what size is the selected part.
+   * @param out_anchors 1D int view denoting where to paste selected part.
+   */
+  void Run(KernelContext &context, const OutImage &out, const Image &pasteFrom,
+           const Coords &in_anchors, const Coords &in_shapes, const Coords &out_anchors) {
+    copyRoi(out, pasteFrom, in_anchors.data[X_AXIS], in_anchors.data[Y_AXIS],
+            in_shapes.data[X_AXIS], in_shapes.data[Y_AXIS],
+            out_anchors.data[X_AXIS], out_anchors.data[Y_AXIS]);
+  }
+
+  void copyRoi(const OutTensorCPU<OutputType> &out, const Image &in, int inXAnchor, int inYAnchor,
+               int inXShape, int inYShape, int outXAnchor, int outYAnchor) {
```

Comment:
```suggestion
  void CopyROI(const OutImage &out. int outXAnchor, int outYAnchor,
               const Image &in, int inXAnchor, int inYAnchor, int inXShape, int inYShape) {
```
I think that output anchors should go next to output and input anchors/shapes next to input.

Reply:
Applied